osquery/tables/system/windows/windows_crashes.cpp

```diff
@@ -0,0 +1,712 @@
+      errorMsg << " DEP access violation.";
+      break;
+    }
+    r["exception_message"] = errorMsg.str();
+  }
+
+  // Log registers from crashed thread
+  auto threadContext =
+      reinterpret_cast<CONTEXT*>(static_cast<unsigned char*>(dumpBase) +
+                                 exceptionStream->ThreadContext.Rva);
+  std::ostringstream registers;
+  // Registers are hard-coded for x64 system b/c lack of C++ reflection on
+  // CONTEXT object
+  registers << "rax:0x" << std::hex << threadContext->Rax;
+  registers << " rbx:0x" << std::hex << threadContext->Rbx;
+  registers << " rcx:0x" << std::hex << threadContext->Rcx;
+  registers << " rdx:0x" << std::hex << threadContext->Rdx;
+  registers << " rdi:0x" << std::hex << threadContext->Rdi;
+  registers << " rsi:0x" << std::hex << threadContext->Rsi;
+  registers << " rbp:0x" << std::hex << threadContext->Rbp;
+  registers << " rsp:0x" << std::hex << threadContext->Rsp;
+  registers << " r8:0x" << std::hex << threadContext->R8;
+  registers << " r9:0x" << std::hex << threadContext->R9;
+  registers << " r10:0x" << std::hex << threadContext->R10;
+  registers << " r11:0x" << std::hex << threadContext->R11;
+  registers << " r12:0x" << std::hex << threadContext->R12;
+  registers << " r13:0x" << std::hex << threadContext->R13;
+  registers << " r14:0x" << std::hex << threadContext->R14;
+  registers << " r15:0x" << std::hex << threadContext->R15;
+  registers << " rip:0x" << std::hex << threadContext->Rip;
+  registers << " segcs:0x" << std::hex << threadContext->SegCs;
+  registers << " segds:0x" << std::hex << threadContext->SegDs;
+  registers << " seges:0x" << std::hex << threadContext->SegEs;
+  registers << " segfs:0x" << std::hex << threadContext->SegFs;
+  registers << " seggs:0x" << std::hex << threadContext->SegGs;
+  registers << " segss:0x" << std::hex << threadContext->SegSs;
+  registers << " eflags:0x" << std::hex << threadContext->EFlags;
+  r["registers"] = registers.str();
+
+  return;
+}
+
+void processDumpMiscInfoStream(Row& r,
+                               MINIDUMP_DIRECTORY* dumpStreamDir,
+                               void* dumpStream,
+                               unsigned long dumpStreamSize) {
+  auto miscInfoStream = static_cast<MINIDUMP_MISC_INFO*>(dumpStream);
+
+  // Log PID, if it exists
+  if (miscInfoStream->Flags1 & MINIDUMP_MISC1_PROCESS_ID) {
+    r["pid"] = BIGINT(miscInfoStream->ProcessId);
+  }
+
+  // Log process times, if they exist
+  if (miscInfoStream->Flags1 & MINIDUMP_MISC1_PROCESS_TIMES) {
+    time_t procTimestamp = miscInfoStream->ProcessCreateTime;
+    struct tm gmt;
+    char timeBuff[64];
+    gmtime_s(&gmt, &procTimestamp);
+    strftime(timeBuff, sizeof(timeBuff), "%Y-%m-%d %H:%M:%S UTC", &gmt);
+    r["process_create_time"] = timeBuff;
+  }
+
+  return;
+}
+
+void processDumpSystemInfoStream(Row& r,
+                                 MINIDUMP_DIRECTORY* dumpStreamDir,
+                                 void* dumpStream,
```

Comment:
Same note about typing. It seems like each of these helper functions follows a rough pattern of:
1.) Coerce the `dumpStream` to the right struct pointer
2.) Get a couple rows
Perhaps this, and maybe other, function(s) could be abstracted into some lambdas and then used inline? Seems like there might be a better design for processing out all of the rows your seeking..